src/deltacode/__init__.py

```diff
@@ -48,14 +49,27 @@ class DeltaCode(object):
     the form of File objects) contained in those scans.
     """
     def __init__(self, new_path, old_path, options):
-        self.new = Scan(new_path)
-        self.old = Scan(old_path)
+        try:
+            self.codebase1 = VirtualCodebase(new_path)
+            self.codebase2 = VirtualCodebase(old_path)
+        except :
+            pass
+        self.new_files_count = 0 #keeps the count of the new file
+        self.old_files_count = 0 #keeps the count of old files
+        self.new_files = [] # a list of [[new file1:Original path],[new file2:Original Path],...]
+        self.old_files = [] # a list of [[old file1:Original path],[old file2:Original Path],...]
+        self.new_files_fingerprint = dict() # map of { {new_file1:fingerprint},{new_file2:fingerprint},...} it will be needed when we need the fingerprints
+        self.old_files_fingerprint = dict() # map of { {old_file1:fingerprint},{old_file2:fingerprint},...}
```

Comment:
What is all this stuff and why is it needed?

Reply:
`self.new_files_count and self.old_files_count` is to keep the track of the new files ,and old files.
I have added  this to make the computation of the statistics easier,otherwise we would have to enumerate the virtual codebase objects to get the cont every time.
`self.new_files_fingerprint and self.old_files_fingerprint` it keeps a mapping of  Resource objects files  path from `codebase1 and codebase2` with respect to the `fingerprints` which would be used in `similarity` comparisons.

Reply:
@majurg  `self.new_files and self.old_files ` it is list of lists comprising of [`Resource objects`,`with their original path`] ,Now we need to keep the track of the original path in the `align_scans` for alignment of the files.

Reply:
Its fine to enumerate on the Virtualcodebase. Stats should be calculated in the end anyway, and optionally for the user. I hate having carrying around all these unneeded fields.

Reply:
Also, codebase objects have counts already that we can use. There is no need to track this twice.

Reply:
@majurg  , yes we can get rid of this `self.new_files_count` and `self.old_files_count` as they are already present in codebase objects  as it is present in the headers of the `codebase objects`, but for the `old files` and `new files` and the `fingerprint`, I think it is better to have the `enumeration` done one time and cache those files in an array, else we will again need to `enumerate` it whenever required.

Reply:
@majurg used the counts from the codebase , removed the additional variables for the `files_count`